### PR TITLE
update collections for essentials

### PR DESCRIPTION
**Pull request by caleb-kaiser**

### CODE REVIEW

File: Queries/Collections/EssentialsByArticle.gql [hunk 1]
```diff
@@ -8,9 +8,7 @@ query EssentialsByArticle(
 		obsessions {
 			nodes {
 				...ObsessionParts
-				essentials(
-					first: 1
-				) {
```
Comment (hallee) [inline Queries/Collections/EssentialsByArticle.gql:13]: Could we keep this? I think there will only ever be 1 essential from the guide/obsession

File: Queries/Collections/EssentialsByArticle.gql [hunk 2]
```diff
@@ -20,9 +18,7 @@ query EssentialsByArticle(
 		guides {
 			nodes {
 				...GuideParts
-				essentials(
-					first: 1
-				) {
```
Comment (hallee) [inline Queries/Collections/EssentialsByArticle.gql:25]: Same here

File: Queries/Fragments/CollectionParts.gql [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+      }
+      sponsor {
+        attribution
+        id
+        logo
+        logoLink
+        name
+        slug
+      }
+    }
```
Comment (hallee) [inline Queries/Fragments/CollectionParts.gql:25]:
```suggestion
	bulletin {
		campaign {
			attribution
			id
			logo
			logoLink
			name
			slug
		}
		clientTracking {
			article
			elsewhere
			logo
		}
		sponsor {
			attribution
			id
			logo
			logoLink
			name
			slug
		}
	}
```

File: Queries/Collections/CollectionById.gql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+query CollectionById(
+	$id: ID!
+) {
+	collectionBy(id: $id) {
```
Comment (hallee) [inline Queries/Collections/CollectionById.gql:4]: Looks like `collectionBy` is deprecated

```suggestion
	collection(id: $id) {
```

File: Queries/Collections/EssentialsByArticle.gql (hunk 1 above)
Comment (caleb-kaiser) [reply Queries/Collections/EssentialsByArticle.gql:13]: Oh haha—that would make things simpler. 

@chriszarate it sounded yesterday as if we were planning on having multiple essentials from guides/obsessions, but it's likely I misunderstood. Can you confirm?

File: Queries/Collections/CollectionById.gql (hunk 4 above)
Comment (chriszarate) [reply Queries/Collections/CollectionById.gql:4]: pass the `idType` arg too for clarity (and so you can avoid the need to create Relay IDs

File: Queries/Collections/EssentialsByArticle.gql (hunk 1 above)
Comment (chriszarate) [reply Queries/Collections/EssentialsByArticle.gql:13]: no plans for that right now

File: Queries/Fragments/CollectionParts.gql (hunk 3 above)
Comment (chriszarate) [reply Queries/Fragments/CollectionParts.gql:25]: would be good to make a fragment (in its own file) for this, too. i believe the type name is `BulletinData`, so

```
fragment BulletinDataParts on BulletinData {
}
```

**Review by hallee — APPROVED** [review]

**Pull request merged**